x-pack/plugins/infra/public/components/logging/log_entry_flyout/log_entry_actions_menu.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButtonEmpty, EuiContextMenuItem, EuiContextMenuPanel, EuiPopover } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React, { useCallback, useMemo, useState } from 'react';
+import url from 'url';
+
+import chrome from 'ui/chrome';
+import { InfraLogItem } from '../../../graphql/types';
+
+const UPTIME_FIELDS = ['container.id', 'host.ip', 'kubernetes.pod.uid'];
+
+export const LogEntryActionsMenu: React.FunctionComponent<{
+  logItem: InfraLogItem;
+}> = ({ logItem }) => {
+  const [hide, isVisible, show] = useVisibility();
+
+  const uptimeLink = useMemo(() => getUptimeLink(logItem), [logItem]);
+
+  const menuItems = useMemo(
+    () => [
+      <EuiContextMenuItem
+        data-test-subj="logEntryActionsMenuItem uptimeLogEntryActionsMenuItem"
+        disabled={!uptimeLink}
+        href={uptimeLink}
+        icon="uptimeApp"
+        key="uptimeLink"
+      >
+        <FormattedMessage
+          id="xpack.infra.logEntryActionsMenu.uptimeActionLabel"
+          defaultMessage="View monitor status"
+        />
+      </EuiContextMenuItem>,
+    ],
+    [uptimeLink]
+  );
+
+  const hasMenuItems = useMemo(() => menuItems.length > 0, [menuItems]);
+
+  return (
+    <EuiPopover
+      anchorPosition="downRight"
+      button={
+        <EuiButtonEmpty
+          data-test-subj="logEntryActionsMenuButton"
+          disabled={!hasMenuItems}
+          iconSide="right"
+          iconType="arrowDown"
+          onClick={show}
+        >
+          <FormattedMessage
+            id="xpack.infra.logEntryActionsMenu.buttonLabel"
+            defaultMessage="Actions"
+          />
+        </EuiButtonEmpty>
+      }
+      closePopover={hide}
+      id="logEntryActionsMenu"
+      isOpen={isVisible}
+      panelPaddingSize="none"
+    >
+      <EuiContextMenuPanel items={menuItems} />
+    </EuiPopover>
+  );
+};
+
+const useVisibility = (initialVisibility: boolean = false) => {
+  const [isVisible, setIsVisible] = useState(initialVisibility);
+
+  const hide = useCallback(() => setIsVisible(false), [setIsVisible]);
+  const show = useCallback(() => setIsVisible(true), [setIsVisible]);
+
+  return [hide, isVisible, show] as [typeof hide, typeof isVisible, typeof show];
+};
+
+const getUptimeLink = (logItem: InfraLogItem) => {
+  const searchExpressions = logItem.fields
+    .filter(({ field, value }) => !!value && UPTIME_FIELDS.includes(field))
```

Comment:
Is `0` a reasonable `value`? If so it will get rejected by `!!0`

New to the code & TS. Please LMK if this is common knowledge or available elsewhere.

Reply:
You're right, it might be a valid value.